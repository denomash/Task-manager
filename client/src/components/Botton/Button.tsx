import classNames from "classnames";
import { IButton } from "../../models/common/IButton";

/**
 * Component which displays a button.
 * @param props Button Properties.
 * @param props.className Button classname.
 * @param props.type Button type.
 * @param props.display Button display.
 * @param props.compact Button compact.
 * @param props.children Button children.
 * @returns The node to render.
 */
const Button = ({
  className,
  type,
  display,
  compact,
  children,
  ...props
}: IButton) => (
  <button
    // eslint-disable-next-line react/button-has-type
    type={type ?? "button"}
    {...props}
    className={classNames(
      "text-white text-sm bg-blue-600 hover:bg-blue-800 font-medium rounded-lg px-5 py-2.5 m-1",
      className,
      {
        "dark:bg-gray-700 dark:hover:bg-gray-600": display === "secondary",
        "dark:bg-red-600 dark:hover:bg-red-700": display === "danger",
      }
    )}
  >
    {children}
  </button>
);

export default Button;
