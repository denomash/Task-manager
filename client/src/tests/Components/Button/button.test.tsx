import { shallow } from "enzyme";

import Button from "../../../components/Botton/Button";

test("Should render button correctly", () => {
  const wrapper = shallow(<Button />);

  expect(wrapper).toMatchSnapshot();
});
