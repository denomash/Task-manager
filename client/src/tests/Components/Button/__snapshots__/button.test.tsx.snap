// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render button correctly 1`] = `ShallowWrapper {}`;
